doc: update documentation with default login details for MetalK8s UI & Grafana service

Ebaneck opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->
'docs'

**Context**: 

See #2351 

**Summary**:

This PR adds the following changes:
- Adds and update the default login credentials for both the MetalK8s GUI and Grafana Service
- Introduces a new section in CSC that deals with changing passwords for local static user
- Removes outdated information in the Account administration page.

The following was not taken care of:
- Add a new landing page png image of the MetalK8s GUI login page(Todo in #2158 )

**Acceptance criteria**: 


---

Closes: #2351
Closes: #2352

Ebaneck committed:
docs: update the default login credentials for MetalK8s UI

Fixes: #2351

Ebaneck committed:
docs: Add procedure to change static user password

bert-e commented:
# Hello ebaneck,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2351-update-documentation-with-default-login-details` into
`development/2.5`
* [w/2.6/bugfix/2351-update-documentation-with-default-login-details](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/2351-update-documentation-with-default-login-details) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

gdemonet commented on docs/operation/account_administration.rst:
```diff
@@ -8,61 +8,35 @@ changing the default username and password for some MetalK8s services.
 
-Administering Grafana
-*********************
+Administering Grafana and MetalK8s GUI
```
I think this doc needs to be rewritten, to show the following sections:

- Dex and OIDC, how it works (connectors, static users...)
- Services relying on Dex (K8s API, Salt API, Grafana, MetalK8s GUI...)
- Procedures to add/edit static users
- Procedures to configure connectors (maybe not filled in right now)

Or, if we really want to put everything under a single "Cluster and Services Configurations" document (I'd vote against this, harder to discover), then maybe just remove this document entirely.

Ebaneck replied:
Done!

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -30,12 +32,11 @@ MetalK8s installation.
    .. code-block:: shell
 
-      root@bootstrap $ htpasswd -bnBC 12 "" <replace-with-password> | tr -d ':\n'
+      root@bootstrap $ htpasswd -bnBC 12 "" <replace-with-password> | tr -d ':'
```
I think the point of this ticket was to not rely on some external package, but build this hash with embedded dependencies...

I'd rather ship a simple Salt module to generate it from the available tooling and document this (in the future, we'll be able to reuse such a module in our CLI).

Ebaneck replied:
In regards to this, out of the box, Salt alone will not work here since the hash is a  `bcrypt` with work factor of atleast 12. Salt does not come with the bcrypt package as checked meaning we need to install this external package.

For a start, we could package bcrypt in the ISO, but we already have httpd-tools added to metalk8s-utils image as from this PR yesterday: https://github.com/scality/metalk8s/pull/2374 which seems sufficient for me given that we have docs on how to run the container

Btw, this ticket is really about updating the documentation to include default login credentials

gdemonet replied:
Oh, right. You're changing the lines on how to provision users, so I thought you were also tackling #2352. If this PR really is only about default creds, then there's a lot of changes that are not related and would need another PR, don't you think?

gdemonet replied:
If we want to use the `metalk8s-utils` container image for generating this, we could just provide the full command to run. Maybe (not tested yet) something like:

```shellscript
$ kubectl --kubeconfig=/etc/kubernetes/admin.conf \
    run -it tmp-gen-pwd \
    --image=metalk8s-registry-from-config.invalid/metalk8s-|release|/metalk8s-utils:|release| \
    --command -- htpasswd -bnBC 12 "" <replace-with-password> | tr -d ':'
```

Ebaneck replied:
I probably will rewrite this PR to include your comments here: https://github.com/scality/metalk8s/pull/2377#discussion_r404694946 then close the two tickets

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -147,6 +148,45 @@ To add a new static user, perform the following operations:
+
+      root@bootstrap $ htpasswd -bnBC 14 "" <replace-with-password> | tr -d ':'
+
+#. From the Bootstrap node, edit the ConfigMap ``metalk8s-dex-config`` and then
+   change the ``hash`` for the selected user:
```
Maybe show an example of the diff on a changed ConfigMap, to make it easier for users?

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -147,6 +148,45 @@ To add a new static user, perform the following operations:
+   the Salt master container and applies salt-states to propagate the new
+   changes down to the underlying services.
+
+   Make sure to replace the <version> prefix in the command below with the
+   currently installed MetalK8s version number e.g 2.5.0
+
+   .. code-block:: shell
+
+      root@bootstrap $ crictl exec -it \
+                         $(crictl ps -q --label io.kubernetes.container.name=salt-master) bash \
+                         -c "salt-run state.sls metalk8s.addons.dex.deployed saltenv=metalk8s-<version>"
```
Just use the current version in docs:

```suggestion
   .. parsed-literal::

      root@bootstrap $ crictl exec -it \
                         $(crictl ps -q --label io.kubernetes.container.name=salt-master) bash \
                         -c "salt-run state.sls metalk8s.addons.dex.deployed saltenv=metalk8s-|release|"
```

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -147,6 +148,45 @@ To add a new static user, perform the following operations:
+
+#. Save and apply the ConfigMap changes.
+
+#. From the Bootstrap node, execute the following command which connects to
+   the Salt master container and applies salt-states to propagate the new
+   changes down to the underlying services.
```
Let's keep it simple:

```suggestion
#. From the Bootstrap node, run the following to propagate the
   changes:
```

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -147,6 +148,45 @@ To add a new static user, perform the following operations:
+                         $(crictl ps -q --label io.kubernetes.container.name=salt-master) bash \
+                         -c "salt-run state.sls metalk8s.addons.dex.deployed saltenv=metalk8s-<version>"
+
+#. Verify that the password has been changed by accessing the related service
+   using the new password.
```
"the related service" is unclear. Let's simply suggest logging in the UI.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

Ebaneck committed:
docs: rewrite account administration with new procedures

bert-e commented:
# History mismatch



Merge commit #2175d306ce6bcf4a080999ef774e555c1fff7cde on the integration branch
`w/2.6/bugfix/2351-update-documentation-with-default-login-details` is merging a branch which is neither the current
branch `bugfix/2351-update-documentation-with-default-login-details` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `bugfix/2351-update-documentation-with-default-login-details` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

NicolasT commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -30,12 +30,19 @@ MetalK8s installation.
    .. code-block:: shell
 
-      root@bootstrap $ htpasswd -bnBC 12 "" <replace-with-password> | tr -d ':\n'
+      root@bootstrap $ crictl exec -it \
```
Nah. The procedure to start the `metalk8s-utils` container (which I'll comment on separately) doesn't fix the Pod/container to a particular node, so `crictl` may not even work.

Also, it's not great form to pass passwords as CLI arguments, they easily end up in shell history and whatnot.

So, instead, why not

```
kubectl run htpasswd-$$ -i --image THE_IMAGE_WE_WANT --restart=Never --rm --command -- sh -c "htpasswd -nBC 12 '' | tr -d ':'"
```

NicolasT replied:
Thinking of it, that won't work with taints.

Now, why don't we install `httpd-utils` on the bootstrap node, and document this package is required (in case someone wants to run this on another node)?

Ebaneck replied:
Ok, i will add the package to the bootstrap node and be done

Ebaneck replied:
Done

NicolasT commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -70,18 +77,14 @@ To add a new static user, perform the following operations:
-      root@bootstrap $ crictl exec -it \
-                         $(crictl ps -q --label io.kubernetes.container.name=salt-master) bash \
-                         -c "salt-run state.sls metalk8s.addons.dex.deployed saltenv=metalk8s-<version>"
+      root@bootstrap $ crictl exec -it \\
```
Always prefer `kubectl ...` over `crictl`.

Ebaneck replied:
fixed!

NicolasT commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -147,6 +150,61 @@ To add a new static user, perform the following operations:
+
+   .. code-block:: shell
+
+      root@bootstrap $ crictl exec -it \
```
See above. Also, de-duplicate this section with the one above. The way to calculate a password hash is always the same.

Ebaneck replied:
Done!

NicolasT commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -147,6 +150,61 @@ To add a new static user, perform the following operations:
+
+   .. parsed-literal::
+
+      root@bootstrap $ crictl exec -it \\
```
`kubectl`

NicolasT commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -223,21 +281,19 @@ perform the following operations:
-                         -c "salt-run state.sls metalk8s.addons.prometheus-operator.deployed saltenv=metalk8s-<version>"
+   .. parsed-literal::
+
+      root@bootstrap $ crictl exec -it \\
```
`kubectl`

NicolasT commented on docs/operation/troubleshooting.rst:
```diff
@@ -9,6 +9,71 @@ after a MetalK8s installation. If you do not find a solution to a problem you
+
+   .. code-block:: shell
+
+      crictl ps -a --label io.kubernetes.container.name=metalk8s-utils
```
`kubectl`, if at all.

NicolasT commented on docs/operation/troubleshooting.rst:
```diff
@@ -9,6 +9,71 @@ after a MetalK8s installation. If you do not find a solution to a problem you
+with the sole purpose to provide a bunch of packages used for troubleshooting
+a cluster. To startup the troubleshooting container, follow this procedure:
+
+#. On the Bootstrap, create a Deployment manifest
```
No, this won't work. When using this image, you'll
- want it to run on a particular node
- likely want to run it in some host namespace(s), potentially with elevated privileges,...

What's below is likely not very useful, except maybe to invoke `htpasswd`, but I think `kubectl run ... --restart=Never -i ...` is a better approach there.

Ebaneck committed:
salt: package & install httpd-tools on Bootstrap node

We need `htpasswd` to generate password hashes for the Dex static
user.
The htpasswd is provided by httpd-tools on redhat based os

Fixes: #2352

thomasdanan commented on docs/installation/services.rst:
```diff
@@ -30,7 +30,13 @@ The login page is loaded, and should resemble the following:
 
 .. image:: img/ui/login.png
 
-Log in with the default login / password (admin / admin).
```
I think we need to change login screenshot

thomasdanan commented on docs/operation/account_administration.rst:
```diff
@@ -3,66 +3,66 @@ Account Administration
+.. note::
 
-#. Now, roll out the new updates for Grafana:
+   The MetalK8s UI and the Grafana service instance make use of the same local
```
maybe ask @lucieleonard to review the sentence

thomasdanan commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -70,18 +75,15 @@ To add a new static user, perform the following operations:
 
 #. Save and apply the changes.
```
isn't it save to apply the change? I mean by saving, the change will be applied right? otherwise, you would need to explain how to apply the change

Ebaneck replied:
Ok, will keep it simple => `Save the changes`

Ebaneck replied:
Done

gdemonet commented on docs/operation/account_administration.rst:
```diff
@@ -3,66 +3,66 @@ Account Administration
---------------------------------------
+While `kube-apiserver` in a MetalK8s cluster enables OpenID Connect(OIDC) as an
+authentication strategy, Dex is configured to provide a static user store and
+a connector based interface to allow MetalK8s users plug-in sub-systems like
+LDAP, SAML, GitHub, Active Directory and others.
```
```suggestion
a connector-based interface which allows MetalK8s users to plug in external
identity providers such as LDAP, SAML, GitHub, Active Directory and others.
```

gdemonet commented on docs/operation/account_administration.rst:
```diff
@@ -3,66 +3,66 @@ Account Administration
 
-Changing Grafana username and password
---------------------------------------
+While `kube-apiserver` in a MetalK8s cluster enables OpenID Connect(OIDC) as an
+authentication strategy, Dex is configured to provide a static user store and
```
Not sure about the contrast you're implying with `While ...`. Maybe:

```suggestion
Kubernetes API enables OpenID Connect (OIDC) as one authentication strategy
(it also supports certificate-based authentication) by trusting Dex as an
OIDC Provider.

Dex can authenticate users against:

 - a static user store (stored in configuration)
 - a connector-based interface, allowing to plug in external ...
```

gdemonet commented on docs/operation/account_administration.rst:
```diff
@@ -3,66 +3,66 @@ Account Administration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-#. Create a file named ``patch-secret.yaml`` that has the following content:
+MetalK8s out of the box enables OpenID Connect(OIDC) based authentication for
```
Parentheses!! 
```suggestion
MetalK8s out of the box enables OpenID Connect (OIDC) based authentication for
```

gdemonet commented on docs/operation/account_administration.rst:
```diff
@@ -3,66 +3,66 @@ Account Administration
+login credentials as: ``admin@metalk8s.invalid`` / ``password``.
 
-   .. code-block:: shell
+The default credentials are defined in the OIDC provider static user store to
```
```suggestion
This default user is defined in Dex configuration as a static user,
to allow...
```

gdemonet commented on docs/operation/account_administration.rst:
```diff
@@ -3,66 +3,66 @@ Account Administration
+.. note::
 
-#. Now, roll out the new updates for Grafana:
+   The MetalK8s UI and the Grafana service instance make use of the same local
+   user store and so changing the default credentials, has a direct impact on
+   both services.
```
Maybe:
```suggestion
   The MetalK8s UI and Grafana are both configured to use OIDC as
   an authentication mechanism, and trust Dex as a Provider. Changing
   the Dex configuration, including the default credentials, will impact
   both UIs.
```

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -30,12 +31,16 @@ MetalK8s installation.
+
+#. Generate a bcrypt hash of your new password.
+
+   - To generate the bcrypt hash, On the Bootstrap node, run the following
```
```suggestion
   - To generate the bcrypt hash, on the Bootstrap node, run the following
```

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -30,12 +31,16 @@ MetalK8s installation.
    .. code-block:: shell
 
-      root@bootstrap $ htpasswd -bnBC 12 "" <replace-with-password> | tr -d ':\n'
+      root@bootstrap $ htpasswd -bnBC 14 "" <password> | tr -d ':'
```
I thought we said the best thing is to never input passwords in clear text on the CLI... So, better use the suggestion from Nicolas:

```suggestion
      root@bootstrap $ htpasswd -nBC 14 "" | tr -d ':'
      New password:
      Re-type new password:
      <your hash here, starting with "$2y$14$">
```

gdemonet commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -68,20 +73,17 @@ To add a new static user, perform the following operations:
-                         -c "salt-run state.sls metalk8s.addons.dex.deployed saltenv=metalk8s-<version>"
+      root@bootstrap $ kubectl exec -n kube-system -c salt-master \\
+                       --kubeconfig /etc/kubernetes/admin.conf \\
+                       salt-master-bootstrap -- bash -c  \\
```
Why the `bash -c` here? We shouldn't need it and be able to just run `salt-run ...` directly.

gdemonet reviewed (changes requested):
Some more changes in the doc, but it's looking much better, thanks a lot!

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2351-update-documentation-with-default-login-details` into
`development/2.5`
* [w/2.6/bugfix/2351-update-documentation-with-default-login-details](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/2351-update-documentation-with-default-login-details) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

* @gdemonet

bert-e commented:
# History mismatch



Merge commit #6752a9c337f00bb9c9fa2b12a02a2dfcd3d7043f on the integration branch
`w/2.6/bugfix/2351-update-documentation-with-default-login-details` is merging a branch which is neither the current
branch `bugfix/2351-update-documentation-with-default-login-details` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `bugfix/2351-update-documentation-with-default-login-details` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2351-update-documentation-with-default-login-details` into
`development/2.5`
* [w/2.6/bugfix/2351-update-documentation-with-default-login-details](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/2351-update-documentation-with-default-login-details) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

* @gdemonet

Ebaneck commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

* @gdemonet







*The following options are set:* **approve**

gdemonet reviewed (approved):
:ok_hand: :shipit:

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue None.

Goodbye ebaneck.
